WordPress/Classes/ViewRelated/Portfolio/PortfolioListViewController.swift

```diff
@@ -0,0 +1,514 @@
+        let deleteHandler = { [weak self] (action: UIAlertAction) in
+            guard let strongSelf = self,
+                let project = strongSelf.projectForObjectID(objectID) else {
+                    return
+            }
+
+            strongSelf.deletePost(project)
+        }
+        let viewHandler = { [weak self] (action: UIAlertAction) in
+            guard let strongSelf = self,
+                let project = strongSelf.projectForObjectID(objectID) else {
+                    return
+            }
+
+            strongSelf.viewPost(project)
+        }
+
+        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        alertController.addCancelActionWithTitle(cancelButtonTitle, handler: nil)
+
+        let filter = filterSettings.currentPostListFilter().filterType
+
+        if filter == .trashed {
+            alertController.addActionWithTitle(publishButtonTitle, style: .default, handler: publishHandler)
+            alertController.addActionWithTitle(draftButtonTitle, style: .default, handler: draftHandler)
+            alertController.addActionWithTitle(deleteButtonTitle, style: .default, handler: deleteHandler)
+        } else if filter == .published {
+            alertController.addActionWithTitle(viewButtonTitle, style: .default, handler: viewHandler)
+            alertController.addActionWithTitle(draftButtonTitle, style: .default, handler: draftHandler)
+            alertController.addActionWithTitle(trashButtonTitle, style: .default, handler: deleteHandler)
+        } else {
+            alertController.addActionWithTitle(viewButtonTitle, style: .default, handler: viewHandler)
+            alertController.addActionWithTitle(publishButtonTitle, style: .default, handler: publishHandler)
+            alertController.addActionWithTitle(trashButtonTitle, style: .default, handler: deleteHandler)
+        }
+
+        WPAnalytics.track(.postListOpenedCellMenu, withProperties: propertiesForAnalytics())
+
+        alertController.modalPresentationStyle = .popover
+        present(alertController, animated: true, completion: nil)
+
+        if let presentationController = alertController.popoverPresentationController {
+            presentationController.permittedArrowDirections = .any
+            presentationController.sourceView = button
+            presentationController.sourceRect = button.bounds
+        }
+    }
+
+    fileprivate func projectForObjectID(_ objectID: NSManagedObjectID) -> Project? {
+
+        var projectManagedObject: NSManagedObject
+
+        do {
+            projectManagedObject = try managedObjectContext().existingObject(with: objectID)
+
+        } catch let error as NSError {
+            DDLogError("\(NSStringFromClass(type(of: self))), \(#function), \(error)")
+            return nil
+        } catch _ {
+            DDLogError("\(NSStringFromClass(type(of: self))), \(#function), Could not find Project with ID \(objectID)")
+            return nil
```

Comment:
You can merge both catch branches. By default, the error is captured, and it's available via the `error` keyword (no need to do anything special for `NSError`).